### PR TITLE
[Issue 163] Removed GODEP references, binaries are lowercase k kraken

### LINKED ISSUE #163: release tarballs for kraken cli tool are now getting created with a capital "K"
This seems to be new. The following list is not exhaustive and I surmise that given the sampling below the issue is systemic in the build now.

`$ for f in {k,g}*gz;do echo $f;tar tvfz $f;done`
```
k2cli_1.1.0_darwin_amd64.tar.gz
-rwxr-xr-x root/root  11862592 2017-09-15 13:35 Kraken
k2cli-release_1.0.9_linux_amd64.tar.gz . # <<--- old build which is lower case
-rwxr-xr-x root/root  11843443 2017-08-25 11:11 k2cli
kraken-rel_1.1.0_linux_amd64.tar.gz
-rwxr-xr-x root/root  11871460 2017-09-15 13:35 Kraken
gox_1.1.0_darwin_amd64.tar.gz
-rwxr-xr-x root/root   3121424 2017-09-15 13:35 Kraken
```

### PR DESCRIPTION
**Pull request by alejandroEsc**
**What this PR does / why we need it**: All external packages are done via vendoring, so removing unnecessary godep references and dependencies since these are no longer required. Created binaries were in caps, no longer the case. Cleaned make file, add a clean function, added more information to `./kraken -v version`
'
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #163

**Special notes for your reviewer**:

**Release note**:

```
NONE
```

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1,326 +0,0 @@
-{
```
Comment (coffeepac) [inline Godeps/Godeps.json]: why is this file being removed?

File: Makefile [hunk 2]
```diff
@@ -1,39 +1,40 @@
...
-godep=GOPATH=$(shell godep path):${GOPATH}
+GOOS       ?= darwin
+GOARCH     ?= amd64
 
+LDFLAGS   := -X github.com/samsung-cnct/kraken/cmd.KrakenMajorMinorPatch=$(VERSION) \
+             -X github.com/samsung-cnct/kraken/cmd.KrakenType=$(TYPE) \
+             -X github.com/samsung-cnct/kraken/cmd.KrakenGitCommit=$(COMMIT) \
 
+build: LDFLAGS += -X github.com/samsung-cnct/kraken/cmd.KrakenlibTag=$(KLIB_VER)
 build:
```
Comment (coffeepac) [inline Makefile]: did you intend for there to be two 'build' targets?

File: Makefile [hunk 3]
```diff
@@ -1,39 +1,40 @@
...
-	@godep go build -ldflags "-X github.com/samsung-cnct/kraken/cmd.KrakenMajorMinorPatch=$(VERSION) \
-		-X github.com/samsung-cnct/kraken/cmd.KrakenType=$(TYPE) \
-		-X github.com/samsung-cnct/kraken/cmd.KrakenGitCommit=$(COMMIT) \
-		-X github.com/samsung-cnct/kraken/cmd.KrakenlibTag=$(KLIB_VER)"
+	@env GOOS=$(GOOS) GOARCH=$(GOARCH) go build -ldflags '$(LDFLAGS)'
 
+compile: LDFLAGS += -X github.com/samsung-cnct/kraken/cmd.KrakenlibTag=$(KLIB_VER)
 compile:
```
Comment (coffeepac) [inline Makefile]: two compile targets, I am unsure what would happen here.

**Comment by alejandroEsc** [comment]
@coffeepac 
Arghh, i was in a rush and forgot to add this in.

It is my understanding that we moved from godep using vendor. That this is no longer important or necessary for managing dependencies.

File: Makefile (hunk 2 above)
Comment (alejandroEsc) [reply Makefile]: As a means to add to the LDFLAGS

https://stackoverflow.com/questions/13919505/how-to-build-multiple-targets-from-one-makefile

Comment (coffeepac) [reply Makefile]: that link does not specify how make will chose either the `build` target with LDFLAGS modified or the `build` target without that variable being modified.  that link states having multiple paths to the same root target `all` and setting different variables by going through `target1` or `target2`.

**Comment by coffeepac** [comment]
godep removal:  are all of our dependencies now vendored?  it appears that way, so hurrah!

please add a comment about this in the PR description.

File: Makefile (hunk 3 above)
Comment (alejandroEsc) [reply Makefile]: @coffeepac 
So the idea here is that this variable needs an additional line that is not part of the install compile target. Here is an example of where that variable is changed:

https://github.com/kubernetes/helm/blob/d7a81c99e18e6cf2b0b666f03d88a8b443b93748/Makefile#L32-L34

And thats what the two compile targets do, allow you to edit that variable and then use it in the following sequence.

**Pull request merged**